hogben/tests/test_simulation.py

```diff
@@ -18,89 +20,103 @@ def sample_structure():
     return sample_1
 
 
-class Test_Simulate():
-    ref = 'OFFSPEC'
-    angle_times = [(0.7, 100, 5), (2.0, 100, 20)]  # (Angle, Points, Time)
+class TestSimulate:
+    angle_times = [(0.3, 100, 1000)]  # (Angle, Points, Time)
     scale = 1
     bkg = 1e-6
     dq = 2
     instrument = 'OFFSPEC'
-    sample_1 = sample_structure()
-    model_1, data_1 = simulate(sample_1, angle_times, scale, bkg, dq, ref)
+
 
     def test_data_streaming(self):
         """Tests that without an input for the datafile, the correct one is picked up"""
-        angle_times = [(0.3, 100, 1000)]
-        _, simulated_datapoints = simulate(self.sample_1, angle_times, self.scale, self.bkg, self.dq, self.ref)
+        sim = Simulation(sample_structure(), self.angle_times, self.scale, self.bkg, self.dq,
```

Comment:
I'll refrain from writing too many comments about the test functions, as they're clearly all quite WIP. But just in case it helps, I got this first unit test to work with the following code:

```
import pytest

import numpy as np
from refnx.reflect import SLD

from unittest import mock
from hogben.simulate import SimulateReflectivity as Simulate



#@pytest.fixture(scope="module")
def sample_structure():
    """Defines a structure describing a simple sample."""
    air = SLD(0, name='Air')
    layer1 = SLD(4, name='Layer 1')(thick=100, rough=2)
    layer2 = SLD(8, name='Layer 2')(thick=150, rough=2)
    substrate = SLD(2.047, name='Substrate')(thick=0, rough=2)

    sample_1 = air | layer1 | layer2 | substrate
    return sample_1


class TestSimulate:
    angle_times = [(0.3, 100, 1000)]  # (Angle, Points, Time)
    scale = 1
    bkg = 1e-6
    dq = 2
    instrument = 'OFFSPEC'


    def test_data_streaming(self):
        """Tests that without an input for the datafile, the correct one is picked up"""
        sim = Simulate(sample_structure(), self.angle_times,
                         self.instrument, self.scale, False)
        simulated_datapoints = sim.simulate()[0]
        np.testing.assert_array_less(np.zeros(len(simulated_datapoints[0])),
                                     simulated_datapoints[3])  # counts

        # Check that the default instrument also works
        sim = Simulate(sample_structure(), self.angle_times,
                         angle_scale=self.scale, polarised=False)
        simulated_datapoints = sim.simulate()[0]
        np.testing.assert_array_less(np.zeros(len(simulated_datapoints[0])),
                                     simulated_datapoints[3])  # counts
```

There were quite some issues, but that's likely the result of the test being written for an earlier development version of the refactor I guess. But again, thought I'd share this just in case that streamlines things a bit.